src/com/google/javascript/jscomp/DependencyOptions.java

```diff
@@ -151,7 +151,34 @@ boolean shouldDropMoochers() {
     return pruneDependencies && dropMoochers;
   }
 
-  Collection<String> getEntryPoints() {
+  Collection<ModuleIdentifier> getEntryPoints() {
     return entryPoints;
   }
+
+  public static final class ModuleIdentifier {
+    private final String name;
+    private final String normalizedModuleName;
+
+    public ModuleIdentifier(String name) {
```

Comment:
Maybe give this factory methods to make code more obvious, e.g. `ModuleIdentifier.forClosureModule("goog:foo.bar")` and `ModuleIdentifier.forFile("foo/bar.js")`?